File: Baekjoon/Baekjoon01240.cpp
// Baekjoon01240.cpp
#include <iostream>

#include <vector>

using namespace std;

using pii = pair<int, int>;

vector<pii> adjList[1'001];

int getDistance(int me, int mom, int target) {
    if (me == target)
        return 0;
    else {
        for (pii& adjV : adjList[me]) {
            if (adjV.first != mom) {
                int result = getDistance(adjV.first, me, target);
                if (result != -1)
                    return result + adjV.second;
            }
        }
    }

    return -1;
}

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(0); cout.tie(0);

    int n, m;
    cin >> n >> m;

    for (int i = 0; i < n - 1; i++) {
        int u, v, d;
        cin >> u >> v >> d;

        adjList[u].push_back({ v, d });
        adjList[v].push_back({ u, d });
    }

    while (m--) {
        int u, v;
        cin >> u >> v;

        cout << getDistance(u, 0, v) << '\n';
    }

    return 0;
}

File: Baekjoon/Baekjoon01240.py
# Baekjoon01240.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())

adjList = [[] for _ in range(n + 1)]
for _ in range(n - 1):
    u, v, d = map(int, input().split())

    adjList[u].append((v, d))
    adjList[v].append((u, d))

def getDistance(me, mom, target):
    if (me == target):
        return 0
    else:
        for son, dis in adjList[me]:
            if (son != mom):
                result = getDistance(son, me, target)
                if (result != -1):
                    return result + dis

    return -1

for _ in range(m):
    u, v = map(int, input().split())
    print(getDistance(u, 0, v))